pallets/xyk-liquidity-mining/src/lib.rs

```diff
@@ -145,6 +144,17 @@ pub mod pallet {
 				Period = PeriodOf<Self>,
 			> + LiquidityMiningInspect<Self::AccountId>;
 
+		/// Oracle source identifier for this pallet.
+		#[pallet::constant]
+		type OracleSource: Get<Source>;
+
+		/// Oracle's liquidity aggregation period.
+		#[pallet::constant]
+		type OraclePeriod: Get<OraclePeriod>;
+
+		/// XYK assets' liquidity oracle.
+		type LiquidityOracle: AggregatedOracle<AssetId, Balance, BlockNumberFor<Self>, pallet_ema_oracle::Price>;
```

Comment:
instead of price from ema oracle pallet. you can use EmaPrice from math package.

this way , you dont need to be dependent on pallet-ema-oracle.